[PyDev/840] Feature: provide test names in configs

rgom opened the pull request:
Provide test names (classes and optionally methods) in (unit)test runner:
1. As part of the created launch configuration, to make them easier to
distinguish.
2. In Run Python unittest configurations:
2a. In Arguments tab
2b. In Interpreter tab

rgom commented:
There are two commits so far - first with writable property, second with read only property.
I prefer the latter, but I'm leaving the former in case it's more valuable for maintainer.

I've added the Group to store label and text, but Group has unneeded border. I couldn't get rid of it, but I think it's not something preventing merge.

fabioz commented:
I just checked it here and I found it a nice addition, however, if there are many tests the description becomes too big and the launch fails because of that... I.e.: use Ctrl+F9 in a module with 20+tests with the default 'test' filter

So, I think it could say something as 'testA, testB + 20 tests' (I'm open to suggestions here), still, the max chars limit for the name has to be respected.

rgom commented:
In the second branch - the one with test configuration from test list - I have tried to alleviate the issue by limiting text length (to some hardcoded value of 128 characters). After exceeding that, I've appended [...] to indicate there are more.
I can rework that if there's other recommended way chosen.
Would you like the following:
- no tests - nothing (already met)
- 1 or 2 tests - their names
- more than 2 tests - names of the first two, followed by the number of the rest
?

fabioz commented:
I think the approach:

no tests - nothing (already met)
1 or 2 tests - their names
more than 2 tests - names of the first two, followed by the number of the rest

seems good, but it should still check the max size just to be sure (who knows the length of the tests that the user uses -- it's entirely possible that the user would exceed the size with a single test).

Also, I usually only keep track of pull requests, so, it'd be nice if you can update this one for that (rebasing is ok until it's merged at master).

rgom commented:
I'm happy that I've provided the implementation which should meet the requirements.
Additionally, I've fixed some minor issues.

If you accept the change, I'll get back to the other branch to rebase it and continue working.

Regards,
Robert

fabioz commented:
Ok, just merged this pull request (in the process I changed the logic for calculating the name a bit as I think that when having big test names the result wasn't ideal -- now it may elide the names of the tests themselves).

Git didn't mark it as merged because In the process I ended up doing a git stash/apply, so, it lost the commits history (it entered as a single commit from me, but I added you as the author in the commit message).

rgom commented:
Thank you for merging and enhancing.

Robert

On Tue, Sep 5, 2017 at 4:28 PM, Fabio Zadrozny <notifications@github.com>
wrote:

> Ok, just merged this pull request (in the process I changed the logic for
> calculating the name a bit as I think that when having big test names the
> result wasn't ideal -- now it may elide the names of the tests themselves).
>
> Git didn't mark it as merged because In the process I ended up doing a git
> stash/apply, so, it lost the commits history (it entered as a single commit
> from me, but I added you as the author in the commit message).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/fabioz/Pydev/pull/207#issuecomment-327192878>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AH2lB8PXq4UfmSWIMOoa6riz_tUA5lCkks5sfVp5gaJpZM4PDEMD>
> .
>
